### PR TITLE
Add WavLM speaker similarity for evaluation

### PR DESCRIPTION
**Pull request by HeCheng0625**
## ✨ Description

The PR add WavLM speaker similarity for evalution, which is widely used for TTS (e.g. Mega-TTS), VC (e.g. LM-VC, UniAudio).

## 👨‍💻 Changes Proposed

- [x] Add `extract_wavlm_similarity in `evaluation/metrics/similarity/wavlm_similarity.py`
- [x] Add `wavlm_similarity` in `METRIC_FUNC`
- [x]  Refine the doc `egs/metrics/README.md`

## 🧑‍🤝‍🧑 Who Can Review?

@lmxue @zhizhengwu @RMSnow 

## ✅ Checklist

- [ ]  Code has been reviewed
- [x]  Code complies with the project's code standards and best practices
- [x]  Code has passed all tests
- [x]  Code does not affect the normal use of existing features
- [x]  Code has been commented properly
- [x]  Documentation has been updated (if applicable)
- [ ]  Demo/checkpoint has been attached (if applicable)

### CODE REVIEW

**Review by RMSnow — APPROVED** [review]
Nice work!

**Review by RMSnow — CHANGES_REQUESTED** [review]
Could you also refine the doc `egs/metrics/README.md`?

File: evaluation/metrics/similarity/wavlm_similarity.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import torch.nn.functional as F
+import os
+import librosa
+import numpy as np
+from tqdm import tqdm
+
+from transformers import Wav2Vec2FeatureExtractor, WavLMForXVector
+
+
+def extract_wavlm_similarity(target_path, reference_path):
```
Comment (lmxue) [inline evaluation/metrics/similarity/wavlm_similarity.py:16]: Could you please add comments for this new function? Adding comments is also a part of the PR checklist.

**Review by lmxue — CHANGES_REQUESTED** [review]
Please add comments for the new function.

**Review by lmxue — APPROVED** [review]

**Review by RMSnow — APPROVED** [review]
Well done!

**Pull request merged**